plugins/modules/net_tools/nmcli.py

```diff
@@ -1078,13 +1089,18 @@ def connection_update(self, nmcli_command):
 
         # Constructing the command.
         for key, value in options.items():
+            if key in self.wifi_sec_secret_options:
+                self.edit_commands += ['set %s %s' % (key, value)]
+                continue
             if value is not None:
                 cmd.extend([key, value])
 
         return self.execute_command(cmd)
 
     def create_connection(self):
         status = self.connection_update('create')
+        if self.edit_commands:
+            status = self.edit_connection()
         if self.create_connection_up:
             status = self.up_connection()
         return status
```

Comment:
What happens if the `self.connection_update('create')` call fails first? My feeling is that there should be an early return in that case. And it's maybe more relevant for `modify_connection` since the edits would then be applied to an existing connection which with old settings.

Reply:
Yeah, I'm not quite sure, I followed the "example" laid out here which would be the same thing:
https://github.com/ansible-collections/community.general/blob/364f49f1f966d8f0a89e98020f4070f8004b8663/plugins/modules/net_tools/nmcli.py#L1104-L1105

Reply:
I think that's less of an issue since bringing up a connection which wasn't created properly should have no effect. (It does hide the actual issue though) Editing however seems like it might leave the connection in an odd state. Again this is probably more of a problem for the `modify_connection` case.

Reply:
Good point, there is a `verify` edit command that might be of use here, but I couldn't parse the response without adding a lot more code.
```
verify   [all | fix]                 :: verify the connection
```
I don't believe it could leave things in a bad state though, what will happen is that the `save` will fail and `quit` will ask for `yes` or `no` and therefore the `nmcli connection edit` command will fail returning a non-zero exit code.

Reply:
Yes, I do see what you mean regarding `modify_connection` though. I'll first check the status before continuing.

Reply:
Alright, I added a potential mitigation. @Ajpantuso, does this look like it will alleviate your concern?
edfb95332de43101d702dca7a35eed0a1b991ec8